File: Week 1/mario-more/mario.c
#include <cs50.h>
#include <stdio.h>

int main(void)
{
    int height = -1;

    // Verifica se a altura está entre 1 e 8
    while (height < 1 || height > 8)
    {
        height = get_int("Height: ");
    }

    // Cada linha da pirâmide
    for (int i = 1; i <= height; i += 1)
    {
        // Pirâmide esquerda (espaços)
        for (int j = 0; j < height - i; j += 1)
        {
            printf(" ");
        }

        // Pirâmide esquerda (hashes)
        for (int j = 0; j < i; j += 1)
        {
            printf("#");
        }

        // Espaço
        for (int j = 0; j < 2; j += 1)
        {
            printf(" ");
        }

        // Pirâmide direita (hashes)
        for (int j = 0; j < i; j += 1)
        {
            printf("#");
        }

        printf("\n");
    }
}

File: Week 6/sentimental-mario-more/mario.py
def get_height():
    while True:
        try:
            height = int(input('Height: '))
        except:
            continue

        if height > 0 and height <= 8:
            return height


def make_half_pyramids(height):
    for i in range(1, height + 1):
        # Pirâmide esquerda (espaços)
        for j in range(height - i):
            print(" ", end='')

        # Pirâmide esquerda (hashes)
        for j in range(i):
            print('#', end='')

        # Espaços
        for j in range(2):
            print(' ', end='')

        # Pirâmide direita (hashes)
        for j in range(i):
            print('#', end='')

        print()


def main():
    height = get_height()
    make_half_pyramids(height)


main()